src/components/views/elements/RoomAliasField.tsx

```diff
@@ -91,16 +107,57 @@ export default class RoomAliasField extends React.PureComponent<IProps, IState>
 
     private validationRules = withValidation({
         rules: [
+            { key: "hasSingleDomain",
+                test: async ({ value }) => {
+                    if (!value) {
+                        return true;
+                    }
+                    // Ignore if we have passed domain
+                    if (this.props.domain) {
+                        return true;
+                    }
+                    const split = value.split(':');
+                    if (split.length < 2) {
+                        return false;
+                    }
+                    return true;
+                },
+                invalid: () => _t("Missing domain separator e.g. (:domain.org)"),
+            },
+            {
+                key: "hasMultipleDomains",
+                test: async ({ value }) => {
+                    if (!value) {
+                        return true;
+                    }
+                    // Ignore if we have passed domain
+                    if (this.props.domain) {
+                        return true;
+                    }
+                    const split = value.split(':');
+                    if (split.length > 2) {
+                        return false;
+                    }
+                    return true;
+                },
+                invalid: () => _t("Multiple domain separators (:) provided, provide an alias with a single domain."),
+            },
             {
                 key: "safeLocalpart",
                 test: async ({ value }) => {
                     if (!value) {
                         return true;
                     }
-                    const fullAlias = this.asFullAlias(value);
-                    // XXX: FIXME https://github.com/matrix-org/matrix-doc/issues/668
-                    return !value.includes("#") && !value.includes(":") && !value.includes(",") &&
-                        encodeURI(fullAlias) === fullAlias;
+                    if (!this.props.domain) {
+                        return true;
+                    } else {
+                        const fullAlias = this.asFullAlias(value);
+                        // XXX: FIXME https://github.com/matrix-org/matrix-doc/issues/668
+                        return !value.includes("#") &&
+                        this.props.domain ? !value.includes(":") : true &&
+                        !value.includes(",") &&
+                            encodeURI(fullAlias) === fullAlias;
```

Comment:
this is a bit painful to read, the ternary could do with going into its own variable

Reply:
Done